lib/pharos/phases/base.rb

```diff
@@ -1,17 +1,42 @@
 # frozen_string_literal: true
 
-require_relative 'logging'
+require 'logger'
 require_relative '../phases'
 
 module Pharos
   module Phases
     class Base
-      include Pharos::Phases::Logging
+      # @return [String]
+      def self.title
+        # XXX: prettier
+        self.name
+      end
 
       def self.register_component(component)
         Pharos::Phases.components << component
       end
 
+      # @param host [Pharos::Configuration::Host]
+      # @param config [Pharos::Config]
+      # @param ssh [Pharos::SSH::Client]
+      # @param master [Pharos::Configuration::Host]
+      def initialize(host, config: nil, ssh: nil, master: nil)
+        @host = host
+        @config = config
+        @ssh = ssh
+        @master = master
+      end
+
+      def logger
+        @logger ||= Logger.new($stdout).tap do |logger|
+          logger.progname = "#{@host}"
+          logger.level = ENV["DEBUG"] ? Logger::DEBUG : Logger::INFO
+          logger.formatter = proc do |_severity, _datetime, progname, msg|
+            "    [%<progname>s] %<msg>s\n" % { progname: progname, msg: msg }
```

Comment:
This needs something to align the `progname` to keep the output between multiple nodes neat.

Reply:
(or stick it in the back)

How about some sort of color pool like in kontena logs? Like use `colors(progname[-1].to_i)` for example.

Reply:
I think if we're already using colors for the `UpCommand` headers (and also the SSH debug output), then it would become a mess... if we wanted to use per-node colors, then IMO we would need to stop using colors for the top-level `UpCommand` output.

The basic INFO-level phase logging is terse enough for this to be readable even without colors, IMO... better to just keep each phase relatively short so the nodes don't get too out of sync.

The SSH `DEBUG` output probably also needs some per-host progname headers... I don't think it's worth the effort to get too fancy with those.

Reply:
I don't see any use for the DEBUG_SSH unless it's flipped like in #117 (the full blown festive  mega-debug thing only with DEBUG_SSH).